kodi_addon_checker/check_py3_compatibility.py

```diff
@@ -0,0 +1,82 @@
+import os
+
+from lib2to3 import refactor
+from tabulate import tabulate
+
+from .report import Report
+from .record import Record, INFORMATION
+
+
+class KodiRefactoringTool(refactor.RefactoringTool):
+
+    def __init__(self, report, *args, **kwargs):
+        self.report = report
+        super(KodiRefactoringTool, self).__init__(*args, **kwargs)
+
+    def short_path(self, path):
+        return os.path.split(path)[1]
+
+    def line_num_for_phrase(self, filepath, phrase):
+        """
+        Return a dictionary having line number of the existing
+        code.
+            :filepath: Path of the file
+            :phrases: List of all the phrases to look for
+
+            :return: A dictionary of type
+                    {'existing_line_of_code': 'line-no'}
+        """
+
+        self.output = {}
+        for p in phrase:
+            with open(filepath, 'r') as f:
+                for (index, line) in enumerate(f):
+                    if p in line:
+                        self.output[p] = (index + 2)
+
+        return self.output
+
+    def print_output(self, old, new, filepath, equal):
+        """
+        Called with the old version, new version, and filepath of a
+        refactored file.
+
+            :old: old text of refactored file
+            :new: new text of refactored file
+            :filepath: Path of the file
+            :equal: Tells whether or not old is equal to new
+        """
+        tabulate.WIDE_CHARS_MODE = False
+        self.headers = ['File', 'Line-No', 'Existing Code', 'Changes required']
+        self.table = []
+
+        if equal:
+            return
+
+        if bool(set(new).intersection(old)):  # Tell whether there is some diff or not
```

Comment:
Isn't that exactly the same what the equal parameter already tells?

Reply:
When I use pdb to check they have different values. The time `equal` was False but the `if bool(set(new).intersection(old))` returns true

Reply:
Please give me an example addon where that happens.

Reply:
Ex: `plugin.video.twitch`

I did `import pdb; pdb.set_trace()` on line above `if equal:` and when the pdb console comes up 
I did `print(equal)` which gives `False` and then doing `bool(set(new).intersection(old))` gives true

Reply:
I thought a bit more about `bool(set(new).intersection(old))` and it doesn't do what you expect.
Try to understand what it really and then you will notice you could also replace it with True.

Your observation doesn't show anything right or wrong. What would you expect?